src/main/java/com/datasonnet/plugins/DefaultYamlFormatPlugin.java

```diff
@@ -0,0 +1,145 @@
+package com.datasonnet.plugins;
+
+/*-
+ * Copyright 2019-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.datasonnet.document.DefaultDocument;
+import com.datasonnet.document.Document;
+import com.datasonnet.document.MediaType;
+import com.datasonnet.document.MediaTypes;
+import com.datasonnet.spi.PluginException;
+import com.datasonnet.spi.ujsonUtils;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.dataformat.yaml.YAMLParser;
+import ujson.Value;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.util.List;
+
+public class DefaultYamlFormatPlugin extends BaseJacksonDataFormatPlugin {
+
+    public static final String DS_PARAM_YAML_HEADER = "removehead";
+    // this may break some things like reading 3.0.0 as a number,
+    // would need to specify this in docs
+    public static final String DS_PARAM_DISABLE_QUOTES = "disablequotes";
+
+    public DefaultYamlFormatPlugin(){
+        supportedTypes.add(MediaTypes.APPLICATION_YAML);
+
+        readerSupportedClasses.add(java.lang.String.class);
+        readerSupportedClasses.add(java.lang.CharSequence.class);
+        readerSupportedClasses.add(java.nio.ByteBuffer.class);
+        readerSupportedClasses.add(byte[].class);
+
+        writerSupportedClasses.add(java.lang.String.class);
+        writerSupportedClasses.add(java.lang.CharSequence.class);
+        writerSupportedClasses.add(java.nio.ByteBuffer.class);
+        writerSupportedClasses.add(byte[].class);
+
+        readerParams.add(DS_PARAM_YAML_HEADER);
+        writerParams.addAll(readerParams);
+        writerParams.add(DS_PARAM_DISABLE_QUOTES);
+
+    }
+
+    @Override
+    public Value read(Document<?> doc) throws PluginException {
+        if (doc.getContent() == null) {
+            return ujson.Null$.MODULE$;
+        }
+
+        try {
+
+            YAMLFactory yaml = new YAMLFactory();
+            ObjectMapper mapper = new ObjectMapper();
```

Comment:
We want to reuse these. Take a look at the json plug-in to see how it's done there.

Reply:
So, I'm not 100% sure how to use the ujson library in this case, since the yaml plugin has support for multiple documents, it essentially creates multiple json objects and creates an array if theres more than one. Because of this, I dont think ujson will parse correctly

Reply:
Sorry, just realized that I meant the Java plugin, that's the one that uses ObjectMapper